sd/etcd/registrar.go

```diff
@@ -69,15 +68,17 @@ func NewRegistrar(client Client, service Service, logger log.Logger) *Registrar
 
 // Register implements the sd.Registrar interface. Call it when you want your
 // service to be registered in etcd, typically at startup.
-func (r *Registrar) Register() {
-	if err := r.client.Register(r.service); err != nil {
+func (r *Registrar) Register() error {
+	err := r.client.Register(r.service)
+	if err != nil {
 		r.logger.Log("err", err)
 	} else {
 		r.logger.Log("action", "register")
 	}
 	if r.service.TTL != nil {
 		go r.loop()
 	}
+	return err
```

Comment:
If Register returns an error, it shouldn't do other things, like `go r.loop`, don't you think?

Reply:
Actually the loop tries to register every TTL, so I feel that I still should let the loop be launched. I don't know how this should be handled.